### PR TITLE
Staging fixing6

### PR DESCRIPTION
**Pull request by Bod9001**
### Changelog:

CL: [Improvement] recharging dock now are wrenched down on start
CL: [Improvement] positional crafting, Eg. using a rolling pin on a peace of dough, the rolled dough will now appear where the dough  was instead of in your inventory
CL: [Improvement] The variable viewer will now highlight constant fields as read-only
CL: [Improvement]  better bandage Interaction text, this should highlight when you successfully heal someone or yourself
CL: [Fix] fixes chat bubbles showing all at once for a long piece of text
CL: [Fix] fixed some Net tabs (UIs) being too big

### CODE REVIEW

File: UnityProject/Assets/Scripts/Items/Medical/HealsTheLiving.cs [hunk 1]
```diff
@@ -77,6 +78,10 @@ public virtual void ServerPerformInteraction(HandApply interaction)
 
 	private void ServerApplyHeal(LivingHealthMasterBase livingHealth, HandApply interaction)
 	{
+		Chat.AddActionMsgToChat(interaction,
+			$"You apply the {this.gameObject.ExpensiveName()} to {livingHealth.gameObject.ExpensiveName()} Healing them.".Color(Color.green),
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Items/Medical/HealsTheLiving.cs]: heal them*

File: UnityProject/Assets/Scripts/Items/Medical/HealsTheLiving.cs [hunk 2]
```diff
@@ -89,6 +94,10 @@ private void ServerSelfHeal(GameObject originator, LivingHealthMasterBase living
 	{
 		void ProgressComplete()
 		{
+			Chat.AddActionMsgToChat(originator,
+				$"You apply the {this.gameObject.ExpensiveName()} to yourself, healing yourself.".Color(Color.green),
+				$"{originator.gameObject.ExpensiveName()} applies the {this.gameObject.ExpensiveName()} to themselves, healing them self.".Color(Color.green)
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Items/Medical/HealsTheLiving.cs:99]: themselves.*

File: UnityProject/Assets/Scripts/Systems/CraftingV2/CraftingIngredient.cs [hunk 3]
```diff
@@ -102,15 +94,50 @@ public bool WillInteract(HandApply interaction, NetworkSide side)
 			return false;
 		}
 
+		public bool WillInteract(InventoryApply interaction, NetworkSide side)
+		{
+			if (HasSimpleRelatedRecipe == false
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Systems/CraftingV2/CraftingIngredient.cs]: split these across multiple one lined if statements so they aren't written this awkwardly.

File: UnityProject/Assets/Scripts/UI/Core/ChatBubble/ChatBubble.cs [hunk 4]
```diff
@@ -160,7 +160,11 @@ public void SetupBubble(Transform _target, string msg, ChatModifier chatModifier
 		QueueMessages(msg, chatModifier);
 
 		cancelSource = new CancellationTokenSource();
-		if (!showingDialogue) StartCoroutine(ShowDialogue(cancelSource.Token));
+		if (!showingDialogue)
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/UI/Core/ChatBubble/ChatBubble.cs]: `showingDialogue == false`

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/UI/Core/ChatBubble/ChatBubble.cs]: use ! only for inversion

**Review by MaxIsJoe — CHANGES_REQUESTED** [review]

File: UnityProject/Assets/Scripts/Items/Medical/HealsTheLiving.cs (hunk 2 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Medical/HealsTheLiving.cs:99]: What makes it not work with the ,?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Medical/HealsTheLiving.cs:99]: it should be , 
my mistake

File: UnityProject/Assets/Scripts/Items/Medical/HealsTheLiving.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Medical/HealsTheLiving.cs]: healing them because you're performing the action

**Review by MaxIsJoe — APPROVED** [review]

**Pull request merged**